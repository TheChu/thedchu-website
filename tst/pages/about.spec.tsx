import { render } from '@testing-library/react';
import { useStaticQuery } from 'gatsby';
import React from 'react';
import uuid from 'short-uuid';
import AboutPage from '../../src/pages/about';

jest.mock('short-uuid');
(uuid.generate as jest.Mock).mockReturnValue('test-uuid');

const mockEdge = (i: number) => ({
  node: {
    description: `image ${i}`,
    localImage: {
      childImageSharp: {
        gatsbyImageData: {
          layout: 'constrained',
          placeholder: {
            fallback: 'test-base64',
          },
          images: {
            fallback: {
              src: 'test-src',
              srcSet: 'test-srcSet',
              sizes: '100vw',
            },
            sources: [],
          },
          width: 1,
          height: 0.6666666666666666,
        },
        original: { src: 'test-src' },
      },
    },
  },
});
const edges = [mockEdge(1), mockEdge(2), mockEdge(3), mockEdge(4)];

(useStaticQuery as jest.Mock).mockReturnValue({
  allAboutImage: {
    edges,
  },
  mastheadImage: {
    localImage: {
      childImageSharp: {
        gatsbyImageData: {
          layout: 'fullWidth',
          placeholder: {
            fallback: 'test-base64',
          },
          images: {
            fallback: {
              src: 'test-src',
              srcSet: 'test-srcSet',
              sizes: '100vw',
            },
            sources: [],
          },
          width: 1,
          height: 0.6666666666666666,
        },
      },
    },
  },
  site: { siteMetadata: { title: 'thedchu' } },
});

describe('AboutPage', () => {
  it('should match AboutPage snapshot', () => {
    const { asFragment } = render(<AboutPage />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render navigation bar', () => {
    const { getByRole } = render(<AboutPage />);
    expect(getByRole('navigation')).toBeVisible();
  });

  it('should render `Scroll down` link', () => {
    const { getByRole } = render(<AboutPage />);
    expect(getByRole('link', { name: 'Scroll down' })).toHaveAttribute(
      'href',
      '/about#about'
    );
  });

  it('should render `About` section', () => {
    const { getByRole } = render(<AboutPage />);
    expect(getByRole('region', { name: 'About' })).toBeVisible();
  });

  it('should render footer', () => {
    const { getByRole } = render(<AboutPage />);
    expect(getByRole('contentinfo')).toBeVisible();
  });
});
