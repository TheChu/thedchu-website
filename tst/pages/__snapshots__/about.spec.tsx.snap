// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutPage should match AboutPage snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar-expand-lg navbar-light fixed-top py-3 "
    id="mainNav"
  >
    <div
      class="container"
    >
      <div
        class="navbar-item"
      >
        <ul
          class="navbar-nav mr-auto my-2 my-lg-0"
        >
          <li
            class="nav-item"
          >
            <a
              class="nav-link nav-link-icon mx-2"
              href="https://instagram.com/thedchu"
            >
              <i
                aria-label="Instagram"
                class="fab fa-instagram"
              />
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link nav-link-icon mx-2"
              href="https://linkedin.com/in/thedchu"
            >
              <i
                aria-label="LinkedIn"
                class="fab fa-linkedin-in"
              />
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link nav-link-icon mx-2"
              href="https://github.com/thechu"
            >
              <i
                aria-label="Github"
                class="fab fa-github"
              />
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link nav-link-icon mx-2"
              href="mailto:chu.david.a@gmail.com"
            >
              <i
                aria-label="Email"
                class="fas fa-envelope"
              />
            </a>
          </li>
        </ul>
      </div>
      <a
        class="navbar-brand navbar-item"
        href="/"
      >
        <svg
          aria-label="DChu"
          class="navbar-brand-logo"
          viewBox="0 0 1000 512"
        >
          <use
            href="test-file-stub#thedchu-logo"
          />
        </svg>
      </a>
      <div
        class="navbar-toggler-container"
      >
        <button
          aria-controls="navbarResponsive"
          aria-expanded="false"
          aria-label="Toggle navigation"
          class="navbar-toggler navbar-toggler-right collapsed"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
      </div>
      <div
        class="collapse navbar-collapse "
        id="navbarResponsive"
      >
        <ul
          class="navbar-nav ml-auto my-2 my-lg-0"
        >
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="/about"
            >
              About
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="#photos"
            >
              Photos
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
  <header
    class="masthead halfHeight gbi--1668593287-test-uuid"
    style="background-position: center; background-repeat: no-repeat; background-size: cover; position: relative; opacity: 0.99;"
  >
    <style>
      
          .masthead.halfHeight.gbi--1668593287-test-uuid:before,
          .masthead.halfHeight.gbi--1668593287-test-uuid:after {
            content: '';
            display: block;
            position: absolute;
            width: 100%;
            height: 100%;
            top: 0;
            left: 0;
            background-color: #040e18;
            transition: opacity 0.5s ease 250ms;
            background-position: center;
background-repeat: no-repeat;
background-size: cover;

          }
          .masthead.halfHeight.gbi--1668593287-test-uuid:before {
            z-index: -100;
            background-image: url(test-base64);
            
            opacity: 1; 
          }
          .masthead.halfHeight.gbi--1668593287-test-uuid:after {
            z-index: -101;
            
            background-image: url(test-base64);
            
          }
        
    </style>
    <div
      class="container h-100"
    >
      <div
        class="row h-100 align-items-center justify-content-center text-center"
      >
        <div
          class="col-lg-10 align-self-end"
        >
          <h1
            class="text-white font-weight-bold"
          >
            About me
          </h1>
        </div>
        <div
          class="col-lg-8 align-self-baseline"
        >
          <p
            class="text-light font-weight-light mb-5"
          >
            Welcome!
          </p>
          <a
            class="btn btn-primary btn-xl js-scroll-trigger"
            href="#about"
          >
            Find Out More
          </a>
        </div>
      </div>
    </div>
  </header>
  <section
    aria-label="About"
    class="bg-light"
    id="about"
  >
    <div
      class="container-fluid p-0"
    >
      <section
        class="about-row"
      >
        <div
          class="image col-lg-6 p-0"
        >
          <div
            class="gatsby-image-wrapper gatsby-image-wrapper-constrained"
            data-gatsby-image-wrapper=""
            style="position: relative; overflow: hidden; display: inline-block; vertical-align: top;"
          >
            <div
              style="max-width: 1px; display: block;"
            >
              <img
                alt=""
                aria-hidden="true"
                role="presentation"
                src="data:image/svg+xml;charset=utf-8,%3Csvg height='0.6666666666666666' width='1' xmlns='http://www.w3.org/2000/svg' version='1.1'%3E%3C/svg%3E"
                style="max-width: 100%; display: block; position: static;"
              />
            </div>
            <img
              alt=""
              aria-hidden="true"
              data-placeholder-image=""
              decoding="async"
              src="test-base64"
              style="height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
            />
            <img
              alt="image 1"
              data-gatsby-image-ssr=""
              data-main-image=""
              data-src="test-src"
              data-srcset="test-srcSet"
              decoding="async"
              loading="lazy"
              sizes="100vw"
              style="height: 100%; left: 0px; position: absolute; top: 0px; transform: translateZ(0); transition: opacity 250ms linear; width: 100%; will-change: opacity; opacity: 0;"
            />
            <noscript />
            <script
              type="module"
            >
              const t="undefined"!=typeof HTMLImageElement&&"loading"in HTMLImageElement.prototype;if(t){const t=document.querySelectorAll("img[data-main-image]");for(let e of t){e.dataset.src&&(e.setAttribute("src",e.dataset.src),e.removeAttribute("data-src")),e.dataset.srcset&&(e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset"));const t=e.parentNode.querySelectorAll("source[data-srcset]");for(let e of t)e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset");e.complete&&(e.style.opacity=1)}}
            </script>
          </div>
        </div>
        <div
          class="content col-lg-6 p-4"
        >
          <p>
            Thanks for checking out my website! I'm David Chu, a software engineer based in Seattle. Born and raised in Panama, I attended college in Southern California before heading up to the Emerald City.
          </p>
        </div>
      </section>
      <section
        class="about-row"
      >
        <div
          class="image col-lg-6 p-0"
        >
          <div
            class="gatsby-image-wrapper gatsby-image-wrapper-constrained"
            data-gatsby-image-wrapper=""
            style="position: relative; overflow: hidden; display: inline-block; vertical-align: top;"
          >
            <div
              style="max-width: 1px; display: block;"
            >
              <img
                alt=""
                aria-hidden="true"
                role="presentation"
                src="data:image/svg+xml;charset=utf-8,%3Csvg height='0.6666666666666666' width='1' xmlns='http://www.w3.org/2000/svg' version='1.1'%3E%3C/svg%3E"
                style="max-width: 100%; display: block; position: static;"
              />
            </div>
            <img
              alt=""
              aria-hidden="true"
              data-placeholder-image=""
              decoding="async"
              src="test-base64"
              style="height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
            />
            <img
              alt="image 2"
              data-gatsby-image-ssr=""
              data-main-image=""
              data-src="test-src"
              data-srcset="test-srcSet"
              decoding="async"
              loading="lazy"
              sizes="100vw"
              style="height: 100%; left: 0px; position: absolute; top: 0px; transform: translateZ(0); transition: opacity 250ms linear; width: 100%; will-change: opacity; opacity: 0;"
            />
            <noscript />
            <script
              type="module"
            >
              const t="undefined"!=typeof HTMLImageElement&&"loading"in HTMLImageElement.prototype;if(t){const t=document.querySelectorAll("img[data-main-image]");for(let e of t){e.dataset.src&&(e.setAttribute("src",e.dataset.src),e.removeAttribute("data-src")),e.dataset.srcset&&(e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset"));const t=e.parentNode.querySelectorAll("source[data-srcset]");for(let e of t)e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset");e.complete&&(e.style.opacity=1)}}
            </script>
          </div>
        </div>
        <div
          class="content col-lg-6 p-4"
        >
          <p>
            As you can probably tell, I'm passionate about photography! I mostly do travel, street, and landscape photography these days. Last year, I switched from a Canon DSLR to a Fujifilm mirrorless camera in favor of portability, and I've been loving it!
          </p>
          <div
            class="gear"
          >
            <p>
              My current gear:
            </p>
            <ul>
              <li>
                Fujifilm X-T4
              </li>
              <li>
                Fujinon XF 23mm F/2 R WR
              </li>
              <li>
                Fujinon XF 50mm F/2 R WR
              </li>
              <li>
                Fujinon XC 50-230mm F/4.5-6.7 OIS II
              </li>
            </ul>
          </div>
        </div>
      </section>
      <section
        class="about-row"
      >
        <div
          class="image col-lg-6 p-0"
        >
          <div
            class="gatsby-image-wrapper gatsby-image-wrapper-constrained"
            data-gatsby-image-wrapper=""
            style="position: relative; overflow: hidden; display: inline-block; vertical-align: top;"
          >
            <div
              style="max-width: 1px; display: block;"
            >
              <img
                alt=""
                aria-hidden="true"
                role="presentation"
                src="data:image/svg+xml;charset=utf-8,%3Csvg height='0.6666666666666666' width='1' xmlns='http://www.w3.org/2000/svg' version='1.1'%3E%3C/svg%3E"
                style="max-width: 100%; display: block; position: static;"
              />
            </div>
            <img
              alt=""
              aria-hidden="true"
              data-placeholder-image=""
              decoding="async"
              src="test-base64"
              style="height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
            />
            <img
              alt="image 3"
              data-gatsby-image-ssr=""
              data-main-image=""
              data-src="test-src"
              data-srcset="test-srcSet"
              decoding="async"
              loading="lazy"
              sizes="100vw"
              style="height: 100%; left: 0px; position: absolute; top: 0px; transform: translateZ(0); transition: opacity 250ms linear; width: 100%; will-change: opacity; opacity: 0;"
            />
            <noscript />
            <script
              type="module"
            >
              const t="undefined"!=typeof HTMLImageElement&&"loading"in HTMLImageElement.prototype;if(t){const t=document.querySelectorAll("img[data-main-image]");for(let e of t){e.dataset.src&&(e.setAttribute("src",e.dataset.src),e.removeAttribute("data-src")),e.dataset.srcset&&(e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset"));const t=e.parentNode.querySelectorAll("source[data-srcset]");for(let e of t)e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset");e.complete&&(e.style.opacity=1)}}
            </script>
          </div>
        </div>
        <div
          class="content col-lg-6 p-4"
        >
          <p>
            I built this website (currently a work in progress!) with: Gatsby, React, GraphQL, Typescript, SASS, AWS S3, hosted on AWS Amplify, and tested with Jest, Testing Library, and Cypress. For now, it'll just be a photography-focused travel/activity blog. Later on I may add in a tech blog or film analysis blog...we'll see!
          </p>
        </div>
      </section>
      <section
        class="about-row"
      >
        <div
          class="image col-lg-6 p-0"
        >
          <div
            class="gatsby-image-wrapper gatsby-image-wrapper-constrained"
            data-gatsby-image-wrapper=""
            style="position: relative; overflow: hidden; display: inline-block; vertical-align: top;"
          >
            <div
              style="max-width: 1px; display: block;"
            >
              <img
                alt=""
                aria-hidden="true"
                role="presentation"
                src="data:image/svg+xml;charset=utf-8,%3Csvg height='0.6666666666666666' width='1' xmlns='http://www.w3.org/2000/svg' version='1.1'%3E%3C/svg%3E"
                style="max-width: 100%; display: block; position: static;"
              />
            </div>
            <img
              alt=""
              aria-hidden="true"
              data-placeholder-image=""
              decoding="async"
              src="test-base64"
              style="height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
            />
            <img
              alt="image 4"
              data-gatsby-image-ssr=""
              data-main-image=""
              data-src="test-src"
              data-srcset="test-srcSet"
              decoding="async"
              loading="lazy"
              sizes="100vw"
              style="height: 100%; left: 0px; position: absolute; top: 0px; transform: translateZ(0); transition: opacity 250ms linear; width: 100%; will-change: opacity; opacity: 0;"
            />
            <noscript />
            <script
              type="module"
            >
              const t="undefined"!=typeof HTMLImageElement&&"loading"in HTMLImageElement.prototype;if(t){const t=document.querySelectorAll("img[data-main-image]");for(let e of t){e.dataset.src&&(e.setAttribute("src",e.dataset.src),e.removeAttribute("data-src")),e.dataset.srcset&&(e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset"));const t=e.parentNode.querySelectorAll("source[data-srcset]");for(let e of t)e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset");e.complete&&(e.style.opacity=1)}}
            </script>
          </div>
        </div>
        <div
          class="content col-lg-6 p-4"
        >
          <p>
            Outside of photography and tech, I play classical guitar, go to the gym, and spend time outdoors. During summer I hike (though that goes hand-in-hand with landscape photography!), and during winter I ski. I'm also a bit of a cinephile and a Real Madrid fan.
          </p>
        </div>
      </section>
    </div>
  </section>
  <footer
    class="bg-dark py-5"
  >
    <div
      class="container"
    >
      <div
        class="row justify-content-center"
      >
        <div
          class="col-lg-8 text-center mt-5 mb-lg-0"
        >
          <a
            class="mx-2"
            href="https://instagram.com/thedchu"
          >
            <i
              aria-label="Instagram"
              class="fab fa-3x mb-3 fa-instagram"
            />
          </a>
          <a
            class="mx-2"
            href="https://linkedin.com/in/thedchu"
          >
            <i
              aria-label="LinkedIn"
              class="fab fa-3x mb-3 fa-linkedin-in"
            />
          </a>
          <a
            class="mx-2"
            href="https://github.com/thechu"
          >
            <i
              aria-label="Github"
              class="fab fa-3x mb-3 fa-github"
            />
          </a>
          <a
            class="mx-2"
            href="mailto:chu.david.a@gmail.com"
          >
            <i
              aria-label="Email"
              class="fas fa-3x mb-3 fa-envelope"
            />
          </a>
        </div>
      </div>
      <div
        class="small text-center text-muted"
      >
        Copyright © 2021 - David Chu
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
